improver/calibration/reliability_calibration.py

```diff
@@ -0,0 +1,388 @@
+        The range 0 to 1 is divided into ranges to give n_probability bins.
+        Note that if single_value_limits is True then two bins will be created
+        with values of 0 and 1 and a width of 1.0E-6. The remaining range will
+        be split into n_probability_bins - 2 ranges.
+
+        Args:
+            n_probability_bins (int):
+                The total number of probability bins required in the
+                reliability tables. If single value limits are turned on, these
+                are included in this total and the minimum number of bins is 3.
+            single_value_limits (bool):
+                Mandates that the extrema bins (0 and 1) should be single
+                valued, with a small precision tolerance of 1.0E-6, e.g. 0 to
+                1.0E-6 for the lowest bin, and 1 - 1.0E-6 to 1 for the highest
+                bin.
+        Returns:
+            numpy.ndarray:
+                An array of 2-element arrays that contain the bounds of the
+                probability bins.
+        Raises:
+            ValueError: If trying to use single_value_limits with 2 or fewer
+                        probability bins.
+        """
+        if single_value_limits:
+            if n_probability_bins <= 2:
+                msg = ("Cannot use single_value_limits with 2 or fewer "
+                       "probability bins.")
+                raise ValueError(msg)
+            n_probability_bins = n_probability_bins - 2
+
+        bin_lower = np.linspace(0, 1, n_probability_bins + 1, dtype=np.float32)
+        bin_upper = np.nextafter(bin_lower, 0, dtype=np.float32)
+        bin_upper[-1] = 1.
+        bins = np.stack([bin_lower[:-1], bin_upper[1:]], 1).astype(np.float32)
+
+        if single_value_limits:
+            fixed_bin_width = 1.0E-6
+            bins[0, 0] = np.nextafter(
+                fixed_bin_width, 1, dtype=np.float32)
+            bins[-1, 1] = np.nextafter(
+                1. - fixed_bin_width, 0, dtype=np.float32)
+            lowest_bin = np.array([0, fixed_bin_width], dtype=np.float32)
+            highest_bin = np.array(
+                [1. - fixed_bin_width, 1], dtype=np.float32)
+            bins = np.vstack(
+                [lowest_bin, bins, highest_bin]).astype(np.float32)
+
+        return bins
+
+    def _create_probability_bins_coord(self):
+        """
+        Construct a dimension coordinate describing the probability bins
+        of the reliability table.
+
+        Returns:
+            iris.coord.DimCoord:
+                A dimcoord describing probability bins.
+        """
+        values = np.mean(self.probability_bins, axis=1, dtype=np.float32)
+        probability_bins_coord = iris.coords.DimCoord(
+            values, long_name='probability_bin', units=1,
+            bounds=self.probability_bins)
+        return probability_bins_coord
+
+    @staticmethod
+    def _create_realiability_table_coords():
```

Comment:
This method and where it is called have a typo:
`_create_realiability_table_coords => _create_reliability_table_coords`

Reply:
Done.